[CI] failure SnapshotResiliencyTests.testSnapshotWithNodeDisconnects

Test failed on 7.x:
```
./gradlew :server:test --tests "org.elasticsearch.snapshots.SnapshotResiliencyTests.testSnapshotWithNodeDisconnects" \
  -Dtests.seed=A01EBDA102D60F73 \
  -Dtests.security.manager=true \
  -Dtests.locale=el-GR \
  -Dtests.timezone=America/Argentina/Rio_Gallegos \
  -Dcompiler.java=12 \
  -Druntime.java=8FIPS \
  -Djavax.net.ssl.keyStorePassword=password \
  -Djavax.net.ssl.trustStorePassword=password
```

Log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/61/console
Build Scans: https://gradle.com/s/sl3v7bw4gytcw


Relevant stack trace
```
java.lang.AssertionError: 
{node3}{zqkXhaXThP}{YPMcpnCCQ9uEo81SwNw_rg}{0.0.0.0}{0.0.0.0:31}{m} should be in published cluster uuid: B7-EwoM3RrmM7MxCDE6isw [committed: true]
version: 12
state uuid: 7OywhP__T_-LCRoaAAAAAA
from_diff: false
meta data version: 6
   coordination_metadata:
      term: 2
      last_committed_config: VotingConfiguration{DqBcnjlIoB,TJpGzKPMZs,clWRuUGfBb,ULJhJdkskv,zqkXhaXThP}
      last_accepted_config: VotingConfiguration{DqBcnjlIoB,TJpGzKPMZs,clWRuUGfBb,ULJhJdkskv,zqkXhaXThP}
      voting tombstones: []
   [test/6BDQSMHPQm60566k4o8tHw]: v[4], mv[1], sv[1], av[1]
      0: p_term [2], isa_ids [Zt21eoaYSFyQV-Kh1Pw_MA]
      1: p_term [2], isa_ids [YJNHgKs1RziAk-yLDsoWqQ]
      2: p_term [2], isa_ids [eMo9NWwsQD2DFrW2rwKnew]
metadata customs:
   index-graveyard: IndexGraveyard[[]]   repositories: org.elasticsearch.cluster.metadata.RepositoriesMetaData@5a3a6a93
nodes: 
   {node0}{DqBcnjlIoB}{sbEo9zcKR2a9pDPjNiscRQ}{0.0.0.0}{0.0.0.0:28}{m}
   {node1}{TJpGzKPMZs}{Zp4Qm14ORTOXwwCOSC6Fdg}{0.0.0.0}{0.0.0.0:29}{m}
   {node2}{ULJhJdkskv}{mmNQymPjSWuWBf6l30Tk2w}{0.0.0.0}{0.0.0.0:30}{m}
   {node3}{zqkXhaXThP}{0D_gE12TRl6s41hIsCetNw}{0.0.0.0}{0.0.0.0:31}{m}, local, master
   {node4}{clWRuUGfBb}{KuZermhbTNSix88GVfeE6w}{0.0.0.0}{0.0.0.0:32}{m}
   {data-node1}{dqeIyubVjP}{pnMps9dkSuS58KAnMsCF3w}{0.0.0.0}{0.0.0.0:34}{d}
routing_table (version 3):
-- index [[test/6BDQSMHPQm60566k4o8tHw]]
----shard_id [test][0]
--------[test][0], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [BewyYcQjOL]], allocation_status[no_valid_shard_copy]]
----shard_id [test][1]
--------[test][1], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [ZTwJxfOUJP]], allocation_status[no_valid_shard_copy]]
----shard_id [test][2]
--------[test][2], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [gwmtjCWjYa]], allocation_status[no_valid_shard_copy]]

routing_nodes:
-----node_id[dqeIyubVjP][V]
---- unassigned
--------[test][2], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [gwmtjCWjYa]], allocation_status[no_valid_shard_copy]]
--------[test][1], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [ZTwJxfOUJP]], allocation_status[no_valid_shard_copy]]
--------[test][0], node[null], [P], recovery_source[existing store recovery; bootstrap_history_uuid=false], s[UNASSIGNED], unassigned_info[[reason=NODE_LEFT], at[2019-07-04T16:30:43.568Z], delayed=true, details[node_left [BewyYcQjOL]], allocation_status[no_valid_shard_copy]]
customs:
   snapshots: SnapshotsInProgress[snapshot]
```

I could not try with  ` -Druntime.java=8FIPS `, but using another Java the test does NOT reproduce for me.

--- a/server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java
+++ b/server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java
@@ -654,7 +654,9 @@ public class SnapshotResiliencyTests extends ESTestCase {
     }
 
     private static ClusterState stateForNode(ClusterState state, DiscoveryNode node) {
-        return ClusterState.builder(state).nodes(DiscoveryNodes.builder(state.nodes()).localNodeId(node.getId())).build();
+        // Remove and add back local node to update ephemeral id on restarts
+        return ClusterState.builder(state).nodes(DiscoveryNodes.builder(
+            state.nodes()).remove(node.getId()).add(node).localNodeId(node.getId())).build();
     }
 
     private final class TestClusterNodes {

